### PR TITLE
#164691218 Users should be able to reset password via email

### PR DESCRIPTION
**Pull request by reifred**
#### What does this PR do?
* Ensure users can reset their passwords via email
#### Description of Task to be completed?
* Send a reset link via users' emails.
* Enable users to reset their passwords in case they forget it.
#### How should this be manually tested?
* After cloning this repo, cd into it and switch to this branch
* Start the development server using ```python manage.py runserver```
* Using Postman test the endpoint below
```
**Reset password endpoint**
POST /api/users/reset-password/
*Sample request body:*
{
	"email": "reifred33@gmail.com"
}
*Sample output on success*
{
    "message": "Please check your email for recovery password link.",
    "status": 201
}
```
* Example mail to receive with a link.
reset
* Use the reset link in the email to reset the password as below.
```
PATCH reset_link
*Sample request body:*
{
   "password": "Pass1234"
}
*Sample output on success*
{
    "message": "Your password has been changed successfully",
    "status": 200
}
```
#### Any background context you want to provide?
If you are using Gmail, you might need to allow [less secure apps](https://myaccount.google.com/lesssecureapps) and [display unlock captcha](https://www.google.com/accounts/DisplayUnlockCaptcha) to make sure you won’t get the error.
#### What are the relevant pivotal tracker stories?
[#164691218 ](https://www.pivotaltracker.com/story/show/164691218)

### CODE REVIEW

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -119,3 +120,70 @@ def get(self, request):
...
         )
+
+
+class ResetPasswordView(APIView):
+    def post(self, request):
+        try:
+            email = request.data["email"]
+            user = User.objects.filter(email=request.data["email"]).first()
+            if not user:
+                return Response({
```
Comment (RonKbS) [inline authors/apps/authentication/views.py]: To protect our users, this check should not be included. Rather, a simple message should be returned of an email being sent. In this way, one will not know which emails we possess in our application when one is not a user of the app

**Review by RonKbS — CHANGES_REQUESTED** [review]
Nice work @reifred.

I request that you change this tests.py file to a `tests` folder and within it, create files for each of this test classes. The format for these files will be like `login_test.py.

File: authors/apps/authentication/tests/test_user_model.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        self.assertNotEqual(previous_count, current_count)
+        self.assertEqual(str(self.user), self.user.email)
+
+    def test_model_returns_fullname_of_user(self):
+        self.user.save()
+        self.assertEqual(self.user.get_full_name, self.user.username)
+
+    def test_model_returns_shortname_of_user(self):
+        self.user.save()
+        self.assertEqual(self.user.get_short_name(), self.user.username)
```
Comment (RonKbS) [inline authors/apps/authentication/tests/test_user_model.py]: @reifred, @NicholusMuwonge, @richien and @Manorlds-Eaglespark,

I should not be pointing out these sort of issues in files.
It speaks of you not having quality work if you raised the PR and the other reviewers not paying attention to detail.
Kindly add a new line here and any other places it is missing.

Comment (reifred) [reply authors/apps/authentication/tests/test_user_model.py]: Am really sorry for such a careless mistake and I have resolved it.

**Review by NicholusMuwonge — APPROVED** [review]
Its really cool work, The feature works fine. I appreciate the efforts @reifred .
I really think your make your code more semantic and readable some more

**Review by RonKbS — APPROVED** [review]

**Pull request merged**